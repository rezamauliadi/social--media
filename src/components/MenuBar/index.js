import React, { Component } from "react";
import PropTypes from "prop-types";

import { Link } from "react-router-dom";
import { Container, Divider, Header, Image, Menu } from "semantic-ui-react";

import getAvatar from "src/helpers/avatar-helper";

const MenuBar = ({ active }) => {
  const menus = [
    { route: "/", name: "Home", key: "home" },
    { route: "/friends", name: "Friends", key: "friends" }
  ];

  const isActive = key => key === active;

  return (
    <div>
      <Container text>
        <Menu style={{ marginTop: "1rem" }} secondary>
          {menus.map((menu, index) => (
            <Menu.Item
              as={Link}
              to={menu.route}
              key={index}
              name={menu.name}
              active={isActive(menu.key)}
              style={{ marginTop: "6px" }}
            />
          ))}
          <Menu.Menu position="right">
            <Menu.Item>
              <Header as="h4">
                <Image src={getAvatar(0)} avatar />
                Hi! You
              </Header>
            </Menu.Item>
          </Menu.Menu>
        </Menu>
      </Container>

      <Divider style={{ marginBottom: "0" }} />
    </div>
  );
};

MenuBar.propTypes = {
  active: PropTypes.string
};

export default MenuBar;
