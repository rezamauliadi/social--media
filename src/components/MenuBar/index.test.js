import React from "react";
import { shallow } from "enzyme";

import MenuBar from "./index.js";

describe("Component: MenuBar", () => {
  it("should match snapshot with active home", () => {
    const component = shallow(<MenuBar active="home" />);

    expect(component).toMatchSnapshot();
  });

  it("should match snapshot with active users", () => {
    const component = shallow(<MenuBar active="users" />);

    expect(component).toMatchSnapshot();
  });
});
