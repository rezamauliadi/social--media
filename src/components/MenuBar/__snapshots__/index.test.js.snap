// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: MenuBar should match snapshot with active home 1`] = `
<div>
  <Container
    text={true}
  >
    <Menu
      secondary={true}
      style={
        Object {
          "marginTop": "1rem",
        }
      }
    >
      <MenuItem
        active={true}
        as={[Function]}
        key="0"
        name="Home"
        style={
          Object {
            "marginTop": "6px",
          }
        }
        to="/"
      />
      <MenuItem
        active={false}
        as={[Function]}
        key="1"
        name="Friends"
        style={
          Object {
            "marginTop": "6px",
          }
        }
        to="/users"
      />
      <MenuMenu
        position="right"
      >
        <MenuItem>
          <Header
            as="h4"
          >
            <Image
              as="img"
              avatar={true}
              src="/images/joe.jpg"
              ui={true}
            />
            Hi! You
          </Header>
        </MenuItem>
      </MenuMenu>
    </Menu>
  </Container>
  <Divider
    style={
      Object {
        "marginBottom": "0",
      }
    }
  />
</div>
`;

exports[`Component: MenuBar should match snapshot with active users 1`] = `
<div>
  <Container
    text={true}
  >
    <Menu
      secondary={true}
      style={
        Object {
          "marginTop": "1rem",
        }
      }
    >
      <MenuItem
        active={false}
        as={[Function]}
        key="0"
        name="Home"
        style={
          Object {
            "marginTop": "6px",
          }
        }
        to="/"
      />
      <MenuItem
        active={true}
        as={[Function]}
        key="1"
        name="Friends"
        style={
          Object {
            "marginTop": "6px",
          }
        }
        to="/users"
      />
      <MenuMenu
        position="right"
      >
        <MenuItem>
          <Header
            as="h4"
          >
            <Image
              as="img"
              avatar={true}
              src="/images/joe.jpg"
              ui={true}
            />
            Hi! You
          </Header>
        </MenuItem>
      </MenuMenu>
    </Menu>
  </Container>
  <Divider
    style={
      Object {
        "marginBottom": "0",
      }
    }
  />
</div>
`;
